### PR TITLE
Fix # parsing bug in string

### LINKED ISSUE #3: Bug when parsing #
tom fails when encountering `#` during parsing.

Here's a minimal reproducible example:

Toml file:
```toml
name = "sketch"
version = "2.2.0"

description = "A CSS-in-Gleam package, made to work with frontend, backend, and directly with lustre!"
internal_modules = ["sketch/internals", "sketch/internals/*"]
licences = ["MIT"]
links = [{title = "Sponsor", href = "https://github.com/sponsors/ghivert"}]
repository = {type = "github", user = "ghivert", repo = "sketch"}

[documentation]
pages = [
  {title = "", path = "#", source = ""},
  {title = "Internal Details", path = "internal-details.html", source = "./docs/internal-details.md"},
  {title = "Usage with BEAM", path = "usage-with-beam.html", source = "./docs/usage-with-beam.md"},
  {title = "Usage with Lustre SSR", path = "usage-with-lustre-ssr.html", source = "./docs/usage-with-lustre-ssr.md"},
]

[dependencies]
gleam_erlang = ">= 0.25.0 and < 1.0.0"
gleam_otp = ">= 0.10.0 and < 1.0.0"
gleam_stdlib = "~> 0.34 or ~> 1.0"
lustre = "~> 4.1"
plinth = "~> 0.2"

[dev-dependencies]
gleeunit = "~> 1.0"
```

Code:

```gleam
import gleam/io
import tom

pub fn main() {
  let file = "content of the file above"
  tom.parse(file)
  |> io.debug()
}
```

Returns:

```
Error(Unexpected("\n", "\""))
```

The line bothering tom is `  {title = "", path = "#", source = ""},`. The bug is not there when removing `#`.

Do you have an idea of why it's happening? Can I help with a PR?

<details><summary>Minimal repro in one file</summary>
<p>

```gleam
import gleam/io
import tom

pub fn main() {
  let file =
    "name = \"sketch\"
version = \"2.2.0\"

description = \"A CSS-in-Gleam package, made to work with frontend, backend, and directly with lustre!\"
internal_modules = [\"sketch/internals\", \"sketch/internals/*\"]
licences = [\"MIT\"]
links = [{title = \"Sponsor\", href = \"https://github.com/sponsors/ghivert\"}]
repository = {type = \"github\", user = \"ghivert\", repo = \"sketch\"}

[documentation]
pages = [
  {title = \"\", path = \"#\", source = \"\"},
  {title = \"Internal Details\", path = \"internal-details.html\", source = \"./docs/internal-details.md\"},
  {title = \"Usage with BEAM\", path = \"usage-with-beam.html\", source = \"./docs/usage-with-beam.md\"},
  {title = \"Usage with Lustre SSR\", path = \"usage-with-lustre-ssr.html\", source = \"./docs/usage-with-lustre-ssr.md\"},
]

[dependencies]
gleam_erlang = \">= 0.25.0 and < 1.0.0\"
gleam_otp = \">= 0.10.0 and < 1.0.0\"
gleam_stdlib = \"~> 0.34 or ~> 1.0\"
lustre = \"~> 4.1\"
plinth = \"~> 0.2\"

[dev-dependencies]
gleeunit = \"~> 1.0\""
  tom.parse(file)
  |> io.debug()
}
```

</p>
</details>

### PR DESCRIPTION
**Pull request by ghivert**
Following my thoughts on what I said in #3, here's a try of bugfixing.

This is working on my example. This can be merged or not as you desire, the idea is to give you better context. I'll let you judge if the code quality is enough or not. (This has been done really quickly. 😅)

### Fixes
- Closes #3

### CODE REVIEW

File: src/tom.gleam [hunk 1]
```diff
@@ -689,14 +689,18 @@ fn skip_whitespace(input: Tokens) -> Tokens {
   }
 }
 
-fn drop_comments(input: Tokens, acc: Tokens) -> Tokens {
-  case input {
-    ["#", ..input] ->
+fn drop_comments(input: Tokens, acc: Tokens, in_string: Bool) -> Tokens {
+  case in_string, input {
```
Comment (lpil) [inline src/tom.gleam:693]: Can you use guard clauses rather than two subjects here please 🙏

**Review by lpil — COMMENTED** [review]
Nice! Could you add tests please

**Comment by ghivert** [comment]
Updated, with a better commit history 🙂

**Review by lpil — APPROVED** [review]
Thank you!!

**Pull request merged**